### PR TITLE
BETSE 0.9.1 bumped.

### PR DESCRIPTION
**Pull request by leycec**
This pull request bumps conda-forge hosting to this project's most recent stable release: **BETSE 0.9.1** (*Luckier Levin*).

## Dear Nordic Gods Above, Why Is This Happening!

This pull request is happening because [cf-regro-autotick-bot](https://github.com/regro/cf-scripts), despite its infinite wisdom, [failed to automatically do so](https://github.com/conda-forge/betse-feedstock/pull/9). Specifically, Circle-CI tests [vomited up an obscure non-human-readable `conda` exception](https://circleci.com/gh/conda-forge/betse-feedstock/19?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```
...
+ make_build_number /home/conda/feedstock_root /home/conda/recipe_root /home/conda/feedstock_root/.ci_support/linux_.yaml
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
Adding in variants from argument_variants
Attempting to finalize metadata for betse
Solving environment: ...working... done
> conda-forge:: No build number clobber gererated!
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/conda_forge_ci_setup/build_utils.py", line 136, in make_build_number
    "Only legacy compilers only valid with build numbers < 1000"
ValueError: Only legacy compilers only valid with build numbers < 1000
+ conda build /home/conda/recipe_root -m /home/conda/feedstock_root/.ci_support/linux_.yaml --clobber-file /home/conda/feedstock_root/.ci_support/clobber_linux_.yaml --quiet
BUILD START: ['betse-0.9.1-py_0.tar.bz2']
export PREFIX=/home/conda/feedstock_root/build_artifacts/betse_1544253421123/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place
export SRC_DIR=/home/conda/feedstock_root/build_artifacts/betse_1544253421123/work
Processing $SRC_DIR
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
  Complete output from command $PREFIX/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-njinqvm9 --no-warn-script-location --no-binary :none: --only-binary :none: --no-index -- setuptools wheel:
  Collecting setuptools
    Could not find a version that satisfies the requirement setuptools (from versions: )
  No matching distribution found for setuptools
  
  ----------------------------------------
Command "/home/conda/feedstock_root/build_artifacts/betse_1544253421123/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-njinqvm9 --no-warn-script-location --no-binary :none: --only-binary :none: --no-index -- setuptools wheel" failed with error code 1 in None
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 456, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 447, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2311, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1477, in build
    cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 374, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 354, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/home/conda/feedstock_root/build_artifacts/betse_1544253421123/work/conda_build.sh']' returned non-zero exit status 1.
Exited with code 1
```

When bumping fails with an egregious spelling mistake (i.e., **"gererated"** – I do *not* believe that is a word), you just know we're in uncharted territory.

## Irrelevant Minutiae that May Actually be Relevant

Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leycec** [comment]
@conda-forge-admin, please rerender

**Comment by leycec** [comment]
**Uh oh.**

@CJ-Wright, @jakirkham: You were both miraculous heroes with [prior PRs](https://github.com/conda-forge/staged-recipes/pull/5574), so... here we are, yet again. Like an unexpectedly ugly mutation of *Red Hiding Hood,* I find myself lost in the woods of [another inscrutable CI error](https://circleci.com/gh/conda-forge/betse-feedstock/20?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

Everyone has more important garbage fires to put out than this. But would either of you happen to know why `conda` fell down and can no longer get up? `setuptools` seems to have vacated the premises, which doesn't quite seem right. Shouldn't *some* sort of `setuptools` distribution always be available? Specifically, I'm seeing the following suspicious lines in the [Circle-CI logfile](https://circleci.com/gh/conda-forge/betse-feedstock/20?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```
> conda-forge:: No build number clobber gererated!
ValueError: Only legacy compilers only valid with build numbers < 1000
```

...and possibly more importantly:

```
  Collecting setuptools
    0 location(s) to search for versions of setuptools:
    Could not find a version that satisfies the requirement setuptools (from versions: )
  Cleaning up...
  Cleaned build tracker '/tmp/pip-req-tracker-vxpc47ww'
  No matching distribution found for setuptools
```

I'm sure this is my fault. Help, for I am like a floundering maiden in the [Forest of Equilibrium](https://www.youtube.com/watch?v=EZkufellkew)!

**Comment by CJ-Wright** [comment]
You might try bumping the build number to 1000 and see if that makes it happy

**Comment by leycec** [comment]
But, that's... crazy. Therefore, it will absolutely work.

**Comment by leycec** [comment]
**Very well, Circle-CI.** The final battle begins. Artificially inflating the build number to 1000 *did* magically silence the [prior "gererated" error,](https://github.com/conda-forge/betse-feedstock/pull/10#issuecomment-446091118) which appears to be good. We now have:

```
> conda-forge:: Build number clobber 1000 -> 1000
+ conda build /home/conda/recipe_root -m /home/conda/feedstock_root/.ci_support/linux_.yaml --clobber-file /home/conda/feedstock_root/.ci_support/clobber_linux_.yaml --quiet
```

Sadly, [the `setuptools` exception remains](https://circleci.com/gh/conda-forge/betse-feedstock/21?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). Is Circle-CI perhaps choking on versioned `setuptools` requirements? We currently list `setuptools >=3.3` as required at both build-time and runtime.

Let's try substituting that with an unversioned `setuptools` requirement and see where we land.

**Comment by leycec** [comment]
Unsurprisingly, that failed in the same manner. Let's revert that, shall we?

**Comment by leycec** [comment]
**So, it comes to this.** While I'm not quite sure I know what's going on, the core issue appears to be that `pip` has been passed the [`--no-index` option](https://pip.pypa.io/en/stable/reference/pip_wheel/#cmdoption-no-index) (preventing `pip` from remotely downloading PyPI-hosted tarballs) but *not* the [`--find-links` option](https://pip.pypa.io/en/stable/reference/pip_wheel/#cmdoption-f) (permitting `pip` to locally access filesystem-hosted tarballs):

```
  Running command $PREFIX/bin/python -m pip install --ignore-installed --no-user
    --prefix /tmp/pip-build-env-5tf7ydmd --no-warn-script-location -v --no-binary :none:
    --only-binary :none: --no-index -- setuptools wheel
  Ignoring indexes: https://pypi.org/simple
  Created temporary directory: /tmp/pip-ephem-wheel-cache-czfzn184
  Re-using requirements tracker '/tmp/pip-req-tracker-t888rqcv'
  Created temporary directory: /tmp/pip-install-8t9pp9q3
  Collecting setuptools
    0 location(s) to search for versions of setuptools:
    Could not find a version that satisfies the requirement setuptools (from versions: )
  Cleaning up...
  Cleaned build tracker '/tmp/pip-req-tracker-t888rqcv'
  No matching distribution found for setuptools  
```

Note the `--no-index` option and *no* corresponding `--find-links` option(s), which produces the `"Ignoring indexes: https://pypi.org/simple"` and `"0 location(s) to search for versions of setuptools:"` lines. So, `pip` is simply unable to find *anything.* This smacks of a critical `conda-build` issue or... something, something.

@CJ-Wright: Is this still my shameful fault or should I try [passing the buck](https://en.wikipedia.org/wiki/Buck_passing) to someone else (i.e., lodge an issue report somewhere else)?

**Comment by CJ-Wright** [comment]
you could go sans `pip` and do a direct `python setup.py install` and see if that does any better. https://gitlab.com/betse/betse#id22

**Comment by leycec** [comment]
Ah! I see. Right. That seems sensible. +1 emoji for trivial solutions. :+1: 

I'm concerned, though, that something might be hideously broken within the conda-forge toolchain. I'm seeing the exact same issue, for example, with a [recent BETSEE bump](https://github.com/conda-forge/betsee-feedstock/pull/3). It stands to reason that other feedstocks might be similarly affected. All of this used to work "sight unseen," as it were. Is this possibly an artifact of the ongoing `conda-build=3` transition?

**_Anyways._** Let's give this simple fix a go. Thanks for being so communicative! You have saved me gratuitous hair-pulling.

**Comment by leycec** [comment]
I'm a bit uncomfortable with these sort of Sudden Installation Exceptions of Death (SIEoD).

I'm sure we're at least partially responsible for the egregious CI failures on display here. Still, there doesn't appear to be a satisfactory explanation of what *exactly* has plummeted off the cliff face of sanity... or why. In fact, I can't even find the conda or conda-forge repository responsible for invoking `pip` with these insane command-line options, which means I can't even submit an issue request to someone more intelligent and/or capable than myself, which means that this is effectively a hard-blocker for us.

Fortunately, the Qt Company recently released [universal platform-portable PySide2 wheels](https://pypi.org/project/PySide2) and the NumPy team similar [universal platform-portable optimized NumPy wheels](https://pypi.org/project/numpy). In theory, this suggests that our PySide2- and NumPy-based application should now be installable via the standard `pip` one-liner: e.g.,

    $ pip install betsee

In light of this blocking conda-forge conundrum *and* improved `pip` support for our core dependencies, we're now considering pivoting back into the PyPI ecosystem – at least temporarily. Until we sort out exactly what's gone belly up, this and a [similar BETSEE pull request](https://github.com/conda-forge/betsee-feedstock/pull/3) will remain on unmerged life support.

We humbly appreciate your fearless assistance, @CJ-Wright. Thanks a bunch! :grapes:

**Comment by dhirschfeld** [comment]
@leycec - this appears to be the same issue affecting my PR https://github.com/conda-forge/trio_asyncio-feedstock/pull/7

My suspicion is that the problem boils down to `click` being installed from defaults because the conda-forge version doesn't have the `gcc7` label applied.

No idea if that's *actually* true but it seems worth trying out. On my PR I've requested someone to apply the `gcc7` label to the `click` package so we can find out.

**Comment by leycec** [comment]
**Oh!** [Exciting gitter conversations](https://gitter.im/conda-forge/conda-forge.github.io?at=5c10e510f4880a60a267cdf9) are lurking in the background. A whole new world of `conda` gossip, hearsay, and speculation is opening before my eyes. In hindsight, I should have just popped my balding head in there first.

I sincerely appreciate your dropping by, @dhirschfeld. We do appear to be stumbling over the [same blocker(s)](https://github.com/conda-forge/trio_asyncio-feedstock/pull/7). So this is <sup>maybe</sup> an unintended side effect of the ongoing compiler transition? (*I knew it.*)

Waiting for benevolent assistance from my betters now appears to be the fix of choice. In the meanwhile, I believe it's time to render our application `pip`-installable. The fateful day has finally come.

Also, this highlights why I can never be a full-fledged conda-forge developer. I *never* would have made the [obscure connections](https://gitter.im/conda-forge/conda-forge.github.io?at=5c11408a8336e22a7d20b814) that @hodgsonw made between Python being downgraded from 3.7 to 3.6, packages including `pip` being installed from the `defaults` rather than `conda-forge` channel, `click` lacking a `gcc7` label, and `pip` being passed contradictory options. Admittedly, I still don't quite see the connection between any of the prior issues and `pip` being passed contradictory options... but it's probably in there. *Somewhere.*

Thanks again, Dave! A random Internet stranger saves my Wednesday. I can now sleep like the dead.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,17 +12,17 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: 1000
```
Comment (CJ-Wright) [inline recipe/meta.yaml]: sorry this should be 0 (I didn't realize that you haven't been migrated yet, we haven't rebuilt `ffmpeg` yet)

Comment (CJ-Wright) [inline recipe/meta.yaml]:
```suggestion
  number: 0
```

**Comment by leycec** [comment]
[**Sweet.**](https://gitter.im/conda-forge/conda-forge.github.io?at=5c11cb625e409525035acb5f) Cheers, @CJ-Wright! The roller coaster of commits begins anew.

File: recipe/meta.yaml (hunk 1 above)
Comment (leycec) [reply recipe/meta.yaml]: No worries! Transitive dependencies are the bane of my Gentoo existence. Let's revert that.

**Comment by CJ-Wright** [comment]
If this doesn't work you might ask for a rerender from the conda-forge-admin bot

**Comment by CJ-Wright** [comment]
@conda-forge-admin please rerender

**Comment by leycec** [comment]
`</sigh>`

Same issue, sadly. The `--no-index` option and lack of a corresponding `--find-links` option still seems to be tripping us up. Do you have any idea where that might be originating from?

**Comment by CJ-Wright** [comment]
I don't.

@conda-forge/core this seems to be running into some odd pip configuration.

```
  No matching distribution found for setuptools
  Exception information:
  Traceback (most recent call last):
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 143, in main
      status = self.run(options, args)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 318, in run
      resolver.resolve(requirement_set)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 102, in resolve
      self._resolve_one(requirement_set, req)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 256, in _resolve_one
      abstract_dist = self._get_abstract_dist_for(req_to_install)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 209, in _get_abstract_dist_for
      self.require_hashes
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/operations/prepare.py", line 218, in prepare_linked_requirement
      req.populate_link(finder, upgrade_allowed, require_hashes)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/req/req_install.py", line 164, in populate_link
      self.link = finder.find_requirement(self, upgrade)
    File "$PREFIX/lib/python3.7/site-packages/pip/_internal/index.py", line 621, in find_requirement
      'No matching distribution found for %s' % req
  pip._internal.exceptions.DistributionNotFound: No matching distribution found for setuptools
  Installing build dependencies: finished with status 'error'
Cleaning up...
  Removing source in /tmp/pip-req-build-87u9hx_k
Removed file://$SRC_DIR from build tracker '/tmp/pip-req-tracker-kqlvygn5'
Removed build tracker '/tmp/pip-req-tracker-kqlvygn5'
Command "/home/conda/feedstock_root/build_artifacts/betse_1544672400732/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-sz7eek2w --no-warn-script-location -v --no-binary :none: --only-binary :none: --no-index -- setuptools wheel" failed with error code 1 in None
Exception information:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 143, in main
    status = self.run(options, args)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 318, in run
    resolver.resolve(requirement_set)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 102, in resolve
    self._resolve_one(requirement_set, req)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 256, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 209, in _get_abstract_dist_for
    self.require_hashes
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/operations/prepare.py", line 298, in prepare_linked_requirement
    abstract_dist.prep_for_dist(finder, self.build_isolation)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/operations/prepare.py", line 109, in prep_for_dist
    "Installing build dependencies"
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/build_env.py", line 122, in install_requirements
    call_subprocess(args, show_stdout=False, spinner=spinner)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/utils/misc.py", line 705, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip._internal.exceptions.InstallationError: Command "$PREFIX/bin/python -m pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-sz7eek2w --no-warn-script-location -v --no-binary :none: --only-binary :none: --no-index -- setuptools wheel" failed with error code 1 in None
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 456, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 447, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2311, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1477, in build
    cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 374, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/utils.py", line 354, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/home/conda/feedstock_root/build_artifacts/betse_1544672400732/work/conda_build.sh']' returned non-zero exit status 1.
Exited with code 1
```

**Comment by ocefpaf** [comment]
> @conda-forge/core this seems to be running into some odd pip configuration.

Can you try to use

```
script: "{{ PYTHON }} setup.py install --single-version-externally-managed --record record.txt"
```

instead of `pip` just to see how it goes?

**Comment by jjhelmus** [comment]
Adding `--no-build-isolation` to the install command fixes the pip configuration issue but why....

Starting with conda-build 3.17.0 the environment variable `PIP_NO_INDEX` is automatically set to True, conda/conda-build#3109.  This is a sensible change, you do not want pip downloading packages from PyPI during the build because those packages should be provided via conda in the build or host environments.

Now in the recipe the host environment provides all the build dependencies for BETSE so there should be no need to download anything from PyPI right?  Well not quite, the BETSE sdist on PyPI includes a `pyproject.toml` file.  Again this is sensible, it follows packaging best practices described in [PEP 518](https://www.python.org/dev/peps/pep-0518/) for specifying build system requirements.  Unfortunately the presence of a `pyproject.toml` file triggers pip to check to see if those dependencies are available, which needs to query PyPI or some other user defined index. Since the package index has been disabled pip fails when checking for the dependencies.

Luckily pip provides the `--no-build-isolation` argument which among other things, skips this check.  When this argument is used the PEP 518 build requirements must already be installed which is done by conda.

**Comment by CJ-Wright** [comment]
Hmm, sounds like the `PIP_NO_INDEX` should prempt the `pyproject.toml` logic.

**Comment by ocefpaf** [comment]
Jeez, what a mess. Looks like a `pip` bug IMO.

**Comment by CJ-Wright** [comment]
Would it be possible to either raise or xref an issue with `pip` about this?

**Comment by ocefpaf** [comment]
> Would it be possible to either raise or xref an issue with `pip` about this?

Did that in the past. It was not a nice experience. I'll leave it to someone with a higher reputation  in the OSS world to do it :grimacing:

**Comment by jjhelmus** [comment]
This is not really a issue with pip, it is doing the right thing (IMHO) when it sees a `pyproject.toml` file.  The presence of the file indicates that the project is aware of [PEP 517](https://www.python.org/dev/peps/pep-0517), so pip does exactly what recommended in the [PEP for a build front end](https://www.python.org/dev/peps/pep-0517/#recommendations-for-build-frontends-non-normative); an isolated build environment is created with the requested build-dependencies and the build backend is called to produce a wheel.  

The `PIP_NO_INDEX` variable disrupted the creation of the isolated build environment.  This is a "good thing", prior to this wheels creation was done in this isolated environment created by pip using package from PyPI rather than in the environments created by conda build.  

conda build should be pro-activate in disabling pip's build isolation, since conda build is already creating such an environment.

**Comment by scopatz** [comment]
So is the real problem that we need to migrate to conda-build >=3.17?

**Comment by jjhelmus** [comment]
> So is the real problem that we need to migrate to conda-build >=3.17?

No versions of conda-build do anything to disable build isolation, until they do `--no-build-isolation` should be added to the install line.

**Comment by scopatz** [comment]
Ahh Ok.

**Comment by jjhelmus** [comment]
> No versions of conda-build do anything to disable build isolation

Sorry, I'm wrong, [`PIP_NO_BUILD_ISOLATION` is being set by conda-build](https://github.com/conda/conda-build/blob/3.17.4/conda_build/build.py#L1859).  Not sure why this isn't having the intended effect.

**Comment by scopatz** [comment]
Does the `env["PIP_NO_BUILD_ISOLATION"] = ""`, because `bool("False") is True`?

**Comment by jjhelmus** [comment]
Let's move the discussion on conda-build behavior to conda/conda-build#3318

**Comment by scopatz** [comment]
Ok!

**Comment by leycec** [comment]
Eat your heart out, Sir Arthur Conan Doyle. Someone give @jjhelmus a bag of venture capital money. Thanks a metric ton to all involved for:

1. Sleuthing through this ponderous conundrum on behalf of @dhirschfeld and I.
2. Cleanly resolving this pull request with a single `pip` option: `--no-build-isolation`.

Yet again, I sing the praises of the conda-forge community. For those following along at home (*hi!*), **[never do this in production code:](https://github.com/conda/conda-build/pull/3319)**

```
        for k, v in env.items():
            if v:  # <---- THIS IS BAD
```

## tl;dr

If your project provides a `pyproject.toml` file and your feedstock repository is similarly affected, simply replace the existing `build/script` command in your feedstock's `recipe/meta.yaml` configuration with the following forward-compatible command:

```
build:
  script: "{{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv"
```

Note the additional `--no-build-isolation` option. Gluten-free organic Christmas pizza for all! :pizza:

**Comment by ocefpaf** [comment]
> This is not really a issue with pip

While I agree with most of the behavior you describe I still think it is an issue with `pip` b/c only the presence of the toml file should not trigger a different behavior. IMHO it should be invoked explicitly via `pip install from_toml` or whatever. (Explicitly is better then implicit, specially when toml files are recommended and people are being surprised by unexpected (AKA I did not read the docs) behavior ;-p)

PS: sorry for answering here, I don't want to prolong a solved problem. Just want to explain why I believe a more explicit behavior in `pip ` invocations would help us all.
